Solutions/Threat Intelligence/Analytic Rules/DomainEntity_DnsEvents.yaml

```diff
@@ -23,56 +23,51 @@ triggerThreshold: 0
-  //Create a list of TLDs in our threat feed for later validation
+  // Define the lookback periods for time-based filters
+  let dt_lookBack = 1h; // Look back 1 hour for DNS events
+  let ioc_lookBack = 14d; // Look back 14 days for threat intelligence indicators
+  // Create a list of TLDs in our threat feed for later validation
   let list_tlds = ThreatIntelligenceIndicator
-  | where TimeGenerated >= ago(ioc_lookBack) and ExpirationDateTime > now()
-  | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
-  | where Active == true  
-  | where isnotempty(DomainName)
-  | extend parts = split(DomainName, '.')
-  | extend tld = parts[(array_length(parts)-1)]
-  | summarize count() by tostring(tld)
-  | summarize make_list(tld);
-  let Domain_Indicators = (union isfuzzy=true
-  (ThreatIntelligenceIndicator
-  | where TimeGenerated >= ago(ioc_lookBack) and ExpirationDateTime > now()
-  | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
-  | where Active == true
-  // Picking up only IOC's that contain the entities we want
-  | where isnotempty(DomainName)
-  | extend TI_DomainEntity = DomainName
-  ),
-  (IoCList
-  | where IoC_Type =~ 'domainname'
-  | where ExpirationDateTime > now()
-  | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IoC
-  | where Active =~ 'True'
-  | extend TI_DomainEntity = IoC
-  | project-away  IoC_Type
-  )
-  );
-  Domain_Indicators
-  // using innerunique to keep perf fast and result set low, we only need one match to indicate potential malicious activity that needs to be investigated
-  | join kind=innerunique (
-    DnsEvents
-    | where TimeGenerated > ago(dt_lookBack)
-    //Extract domain patterns from syslog message
-    | where isnotempty(Name)
-    | extend parts = split(Name, '.')
-    //Split out the TLD
+    | where TimeGenerated >= ago(ioc_lookBack) 
+    | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
+    | where Active == true and ExpirationDateTime > now()
+    | where isnotempty(DomainName)
+    | extend parts = split(DomainName, '.')
     | extend tld = parts[(array_length(parts)-1)]
-    //Validate parsed domain by checking if the TLD is in the list of TLDs in our threat feed
-    | where tld in~ (list_tlds)
-    | extend DNS_TimeGenerated = TimeGenerated
-  ) on $left.TI_DomainEntity==$right.Name
-  | where DNS_TimeGenerated < ExpirationDateTime
-  | summarize DNS_TimeGenerated  = arg_max(DNS_TimeGenerated , *) by IndicatorId, Name, IoC
-  | project DNS_TimeGenerated, Description, ActivityGroupNames, IndicatorId, ThreatType, ExpirationDateTime, ConfidenceScore, Url, Computer, ClientIP, Name, QueryType, Type, IoC, TI_DomainEntity
-  | extend HostName = tostring(split(Computer, '.', 0)[0]), DnsDomain = tostring(strcat_array(array_slice(split(Computer, '.'), 1, -1), '.'))
-  | extend timestamp = DNS_TimeGenerated
+    | summarize count() by tostring(tld)
+    | summarize make_list(tld);
```

Comment:
All make_list operations are required to have max allowed size.

Reply:
Added the max allowed size.